[20849] New `max_message_size` property to limit output datagrams size (#770)
eProsima/Fast-DDS-docs: docs/fastdds/property_policies/non_consolidated_qos.rst

data sizes but should restrict the size of the data it sends to prevent overwhelming network resources
or complying with network traffic policies. The primary attribute for controlling datagram size is
`maxMessageSize`, which sets the upper limit for both the size of datagrams that can be received and
those that can be sent. For applications that need to restrict the size of outgoing datagrams without
changing the size of incoming ones, it's possible to set the new property ``fastdds.max_message_size``.
This property allows for the specific configuration of the maximum number of bytes for datagrams that
are sent. By configuring this property to a value lower than the smallest `maxMessageSize` across all
transports, applications can achieve a lower sending limit while maintaining the ability to receive
larger datagrams.

.. list-table::
   :header-rows: 1
   :align: left

   * - PropertyPolicyQos name
     - PropertyPolicyQos value
     - Default value
   * - ``"fastdds.max_message_size"``
     - ``uint32_t``
     - ``"4294967295"``

.. note::
    An invalid value of ``fastdds.max_message_size`` would log an error,
    and the default value will be used.

> .. tabs::

Reviewer: Add sub-section with title: Setting ``fastdds.max_message_size`` at participant level

    .. tab:: C++

        .. literalinclude:: /../code/DDSCodeTester.cpp
            :language: c++
            :start-after: // MAX_MESSAGE_SIZE_PROPERTY_PARTICIPANT
            :end-before: //!--
            :dedent: 6

    .. tab:: XML

        .. literalinclude:: /../code/XMLTester.xml
            :language: xml
            :start-after: <!-->MAX_MESSAGE_SIZE_PROPERTY_PARTICIPANT<-->
            :end-before: <!--><-->
            :lines: 2,4-16

> .. tabs::

Reviewer: Add sub-section with title: Setting ``fastdds.max_message_size`` at writer level

           :language: xml
           :start-after: <!-->XML-SHM-ENFORCE-META-TRAFFIC
           :end-before: <!--><-->

.. _property_max_message_size:

Maximum Message Size
^^^^^^^^^^^^^^^^^^^^

It is possible to set the maximum number of bytes of an RTPS datagram sent by an RTPSParticipant.

> The value can be set in the RTPSParticipant properties or in the RTPSWriter properties.

Reviewer: Even though the implementation is done at the RTPS level, we should refer to DDS entities here.

We should also add some introduction on the motivation. This property is useful for applications in which the values for the maximum size of a received datagram and a sent datagram should be different.

The configuration attribute in the transport is `maxMessageSize`, which is used both for reception and sending.
If there is an application where we want to receive big datagrams but a lower limit on the sent datagrams is required, then the new property can be used to set that lower limit.

We should explain here that this property will only lower the limit, i.e. it will only have an effect if set to a value lower than the minimum `maxMessageSize` on all transports.

Regarding the structure of the section:
1. Introduction on motivation and how it works (i.e. lower the limit on output datagrams imposed by `maxMessageSize`).
2. The table with name, type, and default value, along with the note of what happens when a wrong value is set.
3. A sub-section on how to set it up in a DomainParticipant.
4. A sub-section on how to set it up in a DataWriter.

Maximum Message Size
^^^^^^^^^^^^^^^^^^^^

One common requirement is the differentiation between the maximum size of received and sent datagrams.
This capability is especially important in scenarios where a system might need to handle large incoming
data sizes but should restrict the size of the data it sends to prevent overwhelming network resources
or complying with network traffic policies. The primary attribute for controlling datagram size is
`maxMessageSize`, which sets the upper limit for both the size of datagrams that can be received and

> those that can be sent. For applications that need to restrict the size of outgoing datagrams without
> changing the size of incoming ones, it's possible to set the new property ``fastdds.max_message_size``.

Reviewer: ```suggestion
those that can be sent.
Property ``fastdds.max_message_size`` allows restricting the size of outgoing datagrams without
changing the size of incoming ones.
```

This property allows for the specific configuration of the maximum number of bytes for datagrams that
are sent. By configuring this property to a value lower than the smallest `maxMessageSize` across all
transports, applications can achieve a lower sending limit while maintaining the ability to receive

> larger datagrams.

Reviewer: Line breaks after full stops.
